### PR TITLE
easyrace

### PR DESCRIPTION
**Pull request by DarkSlash88**
Added the easyrace code, which is still tierless, but bans most annoying specials, and removes MIABs from chests

### CODE REVIEW

File: chestrandomizer.py [hunk 1]
```diff
@@ -322,6 +322,8 @@ def mutate_contents(self, guideline=None, monster=None,
         elif monster is False:
             chance += 3
             chance = min(chance, 50)
+        if easyrace is True:
+		    chance = random.randint(4, 50)
```
Comment (42B) [inline chestrandomizer.py]:
```python
if easyrace:  # ?
```
Also, the following line seems to be indented too far?

File: monsterrandomizer.py [hunk 2]
```diff
@@ -1569,6 +1569,9 @@ def mutate_special(self, darkworld=False, narshesafe=False):
         elif branch > branches[1]:
             # bonus special
             valid = set(range(10, 0x1F))
+            if saferace is True:
```
Comment (42B) [inline monsterrandomizer.py]:
```python
if saferace:  # ?
```

File: randomizer.py [hunk 3]
```diff
@@ -2719,7 +2720,7 @@ def manage_colorize_animations():
 def manage_items(items, changed_commands=None):
     from itemrandomizer import (set_item_changed_commands, extend_item_breaks)
     always_break = True if "collateraldamage" in activated_codes else False
-    crazy_prices = True if "madworld" in activated_codes else False
+    crazy_prices = True if "madworld" or "easyrace" in activated_codes else False
```
Comment (42B) [inline randomizer.py]: Wouldn't this line always return True?

```python
>>> bool("madworld")
True
```

File: randomizer.py [hunk 4]
```diff
@@ -3120,7 +3121,7 @@ def ensure_striker():
 
 
 def manage_chests():
-    crazy_prices = True if "madworld" in activated_codes else False
+    crazy_prices = True if "madworld" or "easyrace" in activated_codes else False
```
Comment (42B) [inline randomizer.py]: Same problem as above?

File: randomizer.py [hunk 5]
```diff
@@ -3639,7 +3640,7 @@ def get_shops():
 def manage_shops():
     buyables = set([])
     descriptions = []
-    crazy_shops = True if "madworld" in activated_codes else False
+    crazy_shops = True if "madworld" or "easyrace" in activated_codes else False
```
Comment (42B) [inline randomizer.py]: ?

File: randomizer.py [hunk 6]
```diff
@@ -5268,7 +5269,7 @@ def challenge_battle_validator(formation):
                     and f.get_music() != 0]):
                 return False
         best_drop = formation.get_best_drop()
-        if best_drop and (best_drop.price <= 2 or best_drop.price >= 30000 or "madworld" in activated_codes):
+        if best_drop and (best_drop.price <= 2 or best_drop.price >= 30000 or "madworld" or "easyrace" in activated_codes):
```
Comment (42B) [inline randomizer.py]: ?

File: randomizer.py [hunk 7]
```diff
@@ -6077,7 +6078,7 @@ def manage_spookiness():
         nowhere_to_run_bottom_sub.write(fout)
         
 def manage_dances():
-    if 'madworld' in activated_codes:
+    if 'madworld' or 'easyrace' in activated_codes:
```
Comment (42B) [inline randomizer.py]: ?

File: randomizer.py [hunk 8]
```diff
@@ -6402,7 +6404,7 @@ def randomize():
         except:
             multiplier = None
         set_randomness_multiplier(multiplier)
-    elif 'madworld' in activated_codes:
+    elif 'madworld' or 'easyrace' in activated_codes:
```
Comment (42B) [inline randomizer.py]: ?

File: randomizer.py [hunk 9]
```diff
@@ -6434,11 +6436,12 @@ def randomize():
     reseed()
 
     spells = get_ranked_spells(sourcefile)
-    if 'madworld' in activated_codes:
+    if 'madworld' or 'easyrace' in activated_codes:
```
Comment (42B) [inline randomizer.py]: ?

File: chestrandomizer.py (hunk 1 above)
Comment (DarkSlash88) [reply chestrandomizer.py]: You're right. Let me check the source.

File: randomizer.py (hunk 3 above)
Comment (DarkSlash88) [reply randomizer.py]: Why would it always be true? madworld or easyrace won't always be in activated codes

Comment (DarkSlash88) [reply randomizer.py]: Oooh nevermind I understand. Thanks!

File: monsterrandomizer.py (hunk 2 above)
Comment (DarkSlash88) [reply monsterrandomizer.py]: What do you mean?

**Pull request closed without merging**